dandi/dandiapi.py

```diff
@@ -450,53 +452,29 @@ def iter_upload(self, dandiset_id, version_id, asset_metadata, filepath):
             lgr.debug("Uploading %s in %d parts", filepath, len(parts))
             with storage.session():
                 with open(filepath, "rb") as fp:
-                    for part, etag_part in zip(parts, etagger.get_parts()):
-                        if part["size"] != etag_part.size:
-                            raise RuntimeError(
-                                f"Server and client disagree on size of upload part"
-                                f" {part['part_number']}; server says"
-                                f" {part['size']}, client says {etag_part.size}"
+                    with ThreadPoolExecutor() as executor:
```

Comment:
I think that default behavior of "If max_workers is None or not given, it will default to the number of processors on the machine, multiplied by 5" would be way too much on many contemporary boxes - we do not want to start 300 parallel uploads for a file.  I think we should
- add `jobs=5` kwarg here 
- percolate its specification to the `dandi upload`.  apparently we do not even have `--jobs` for upload (only for `download`) and rely on `pyout`'s default instead of specifying `max_workers` to it.  So we need to add `--jobs` to pass max_workers to `LogSafeTabular` but also somehow allow for specification of number of jobs per file upload.  Could be separate `--jobs-per-file` (or `--threads-per-file`?) or make `--jobs` to  support `[jobs][:threads]` specification so that user could specify e.g. `--jobs 2:10` which would instruct 2 files processed in parallel with 10 threads for upload per file.

WDYT @jwodder would be the best way to expose this "multiplicity" of `--jobs`?

Reply:
@yarikoptic I prefer the `--jobs 2:10` syntax.

Reply:
@yarikoptic `--jobs` option added.